### ISSUE TITLE
/doc deletes existing functions

### ISSUE THREAD

**Issue by haukened** [microsoft/vscode-copilot-release#1476]
- Copilot Chat Extension Version: v1.222.0
- VS Code Version: Version: 1.92.1 (Universal)
- OS Version: Darwin arm64 23.6.0

Steps to Reproduce:

1. Write a function
2. Highlight the function
3. Trigger in-line CoPilot `/doc` function
4.  `/doc` overwrites the code you've already written

Example:

Before `/doc`

```
func (c *DNSCache) Get(key string) (string, bool) {
	// check the cache for a value
	value, ok := c.entries[key]
	// if the value is not in the cache, return false
	if !ok {
		return "", false
	}
	// if the value is expired, delete it from the cache and return false
	if value.Expires.Before(time.Now()) {
		delete(c.entries, key)
		return "", false
	}
	// return the value
	return value.Value, true
}
```

After `/doc`

```
// Get retrieves the value associated with the given key from the DNS cache.
// It returns the value and a boolean indicating whether the value was found in the cache.
// If the value is not in the cache or if it has expired, it returns an empty string and false.
func (c *DNSCache) Get(key string) (string, bool) {
	// ...
}
	// return the value
	return value.Value, true
}
```

**Comment by canuckjacq** [comment]
Same thing? https://github.com/microsoft/vscode-copilot-release/issues/1479

**Comment by haukened** [comment]
Yep, I just didn't think to search  "Eats Lines" when I was searching for a /doc bug.

**Comment by BenjaminHoegh** [comment]
Great to see I'm not alone with the problem

**Comment by morgano86** [comment]
I'm having the same issue, very frustrating as it works occasionally.

**Comment by ryanbooker** [comment]
It's pretty difficult to do any of the learning exercises and presumably the cert exam with this issue.

**Comment by FelixLisczyk** [comment]
I'm experiencing the same issue with the inline chat feature when generating documentation comments for an entire file. Existing code and comments are often replaced with placeholders such as `// ... existing code ...` or `// Implementation goes here`. GitHub Copilot should either instruct the model to provide the complete code without placeholders or improve its processing to accurately identify which parts of the file to replace. This feature works much better in [Cursor](cursor.com).

**Comment by jhughes-dev** [comment]
Yeah, this is pretty bad. It should never delete code.

**Comment by ArmandBurger** [comment]
Also experiencing the // ... existing code ... bug. Makes it very difficult to use the feature, especially on linux which does not have a win+v like copy history, so I have to manually copy over the doc comments to another file, cancel the copilot edit and then paste the doc into the correct places.

**Comment by AdamSimpson** [comment]
Such an impactful issue being open for so long is surprising. Anyone have any workarounds?

**Comment by the-color-cyan** [comment]
Just tried testing documentation generation with Copilot today and this still seems to be happening, even across several different models.
